### PR TITLE
[Product CRUD] Abstract todos

### PR DESCRIPTION
**Pull request by mikejolley**
Handles misc methods in abstract class which were marked todo.

- Downloads
- Ratings

And fixes some things I noticed during coding.

### CODE REVIEW

File: includes/abstracts/abstract-wc-legacy-product.php
Comment (justinshreve) [inline includes/abstracts/abstract-wc-legacy-product.php]: Hopefully I'm not just seeing things this time (:P), but it looks like there is an extra space in front.

File: includes/abstracts/abstract-wc-product.php
Comment (justinshreve) [inline includes/abstracts/abstract-wc-product.php]: `$variation` -> `$this`

File: includes/class-wc-comments.php
Comment (justinshreve) [inline includes/class-wc-comments.php]: int

File: includes/wc-deprecated-functions.php
Comment (justinshreve) [inline includes/wc-deprecated-functions.php]: Should be `$argument` instead of `$function`

**Review by emdashcodes — APPROVED** [review]
Left 4 minor comments. Otherwise, I think this can be merged after those are fixed.

**Comment by justinshreve** [comment]
Based on the size of the incoming data store PR, I think it might make sense to hold off and have us rebase this one after it's merged. This has a few changes that will be incompatible, but I think it will be easier to rebase this one then the other.

**Pull request merged**
